daft/daft.pyi

```diff
@@ -480,6 +483,14 @@ class S3Config:
         """Creates an S3Config, retrieving credentials and configurations from the current environment"""
         ...
 
+class S3Credentials:
+    key_id: str
+    access_key: str
+    session_token: str | None
+    expiry: int | None
```

Comment:
This should probably be a datetime or some kind of timestamp?

Reply:
I kept it as an integer because the pyo3 version we use does not support datetime with abi3. What I think I'll do is convert datetime to unix timestamp in python for now, and then change it when we upgrade pyo3 to 0.21 (#2288). To the user, it should be the same usage, just passing in a datetime object